internal/job/checkout.go

```diff
@@ -42,6 +42,12 @@ func (e *Executor) configureGitCredentialHelper(ctx context.Context) error {
 		return fmt.Errorf("configuring git credential.helper: %v", err)
 	}
 
+	// so that the new credential helper will be used for all github urls
+	err = e.shell.RunWithoutPrompt(ctx, "git", "config", "--global", "url.https://github.com/.insteadOf", "git@github.com:")
+	if err != nil {
+		return fmt.Errorf("enabling git rewrite: %v", err)
+	}
+
```

Comment:
Should we add something like

```suggestion
	// Don't ssh keyscan - we won't be using ssh anyway.
	e.SSHKeyscan = false
	
	// so that the new credential helper will be used for all github urls
	err = e.shell.RunWithoutPrompt(ctx, "git", "config", "--global", "url.https://github.com/.insteadOf", "git@github.com:")
	if err != nil {
		return fmt.Errorf("enabling git rewrite: %v", err)
	}

```

it means that users running on an instance without ssh installed (which might be compute instances?) won't see errors, and we know that YAGNI at this point

Reply:
Maybe, but then on self-hosted, if there is an SSH key for a non-GitHub SCM, and this is the first checkout of that repo on that machine, then we would fail with an unauthorized host.

I think the self-hosted situation will require significantly more thought. So i'm going to scatter some to-do comments throughout.

Reply:
Ok done.